src/js/d3Utils.js

```diff
@@ -23,6 +23,42 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
         return d3.selectAll(elem.toArray());
     };
 
+    /* Given the object "d", returns value of:
+     * - "id" if it exists on the top-level object,
+     * - "data.id" if it exists on a "data" object
+     * - "undefined" if neither of these exist
+     */
+    floe.d3.idExtractor = function (d) {
+        try {
+            var id = d.id !== undefined ? d.id : d.data.id;
+            return id !== undefined ? id : undefined;
+        } catch (e) {
+            return undefined;
+        }
+    };
+
+    // Given a selection of D3 elements and an ID, returns only the elements
+    // matching that ID
+    floe.d3.filterById = function(d3Selection, currentlyPlayingDataId) {
```

Comment:
@amb26 I've done a first pass at refactoring (still working on some of the pieces) to a model-level approach for everything except the D3-specific functionality that's needed to filter and select DOM elements based on D3's own model of binding data to DOM elements. I think this fits with what we all talked about last year around how and when we make use of D3 in these components.